sklearn/model_selection/tests/test_split.py

```diff
@@ -985,6 +986,43 @@ def test_label_kfold():
                          next, LabelKFold(n_splits=3).split(X, y, labels))
 
 
+def test_homogeneous_time_series_cv():
+    X = [[1, 2], [3, 4], [5, 6], [7, 8], [9, 10], [11, 12], [13, 14]]
+
+    # Should fail if there are more folds than samples
+    assert_raises_regexp(ValueError, "Cannot have number of folds.*greater",
+                         next,
+                         HomogeneousTimeSeriesCV(n_splits=9).split(X))
```

Comment:
As a corner case, you might have to check for when n_splits is set to the number of samples.